include/deal.II/dofs/dof_accessor.templates.h

```diff
@@ -218,1035 +215,285 @@ namespace internal
-          dof_handler, obj_index, fe_index, local_index, global_index);
-      }
-
 
-
-      template <int spacedim>
-      static types::global_dof_index
-      get_dof_index(const dealii::DoFHandler<3, spacedim> &dof_handler,
-                    const unsigned int                     obj_level,
-                    const unsigned int                     obj_index,
-                    const unsigned int                     fe_index,
-                    const unsigned int                     local_index,
-                    std::integral_constant<int, 3>)
-      {
-        return dof_handler.levels[obj_level]->dof_object.get_dof_index(
-          dof_handler, obj_index, fe_index, local_index);
-      }
-
-
-      template <int spacedim>
-      static void
-      set_dof_index(const dealii::DoFHandler<3, spacedim> &dof_handler,
-                    const unsigned int                     obj_level,
-                    const unsigned int                     obj_index,
-                    const unsigned int                     fe_index,
-                    const unsigned int                     local_index,
-                    std::integral_constant<int, 3>,
-                    const types::global_dof_index global_index)
-      {
-        dof_handler.levels[obj_level]->dof_object.set_dof_index(
-          dof_handler, obj_index, fe_index, local_index, global_index);
-      }
-
-
-      template <int spacedim>
-      static types::global_dof_index
-      get_dof_index(const dealii::hp::DoFHandler<1, spacedim> &dof_handler,
-                    const unsigned int                         obj_level,
-                    const unsigned int                         obj_index,
-                    const unsigned int                         fe_index,
-                    const unsigned int                         local_index,
-                    const std::integral_constant<int, 1> &)
-      {
-        return dof_handler.levels[obj_level]->get_dof_index(obj_index,
-                                                            fe_index,
-                                                            local_index);
-      }
-
-
-      template <int spacedim>
+      /**
+       * Set the @p local_index-th degree of freedom corresponding to the
+       * finite element specified by @p fe_index on the vertex with global
+       * number @p vertex_index to @p global_index.
+       */
+      template <typename DoFHandlerType, int d>
       static void
-      set_dof_index(const dealii::hp::DoFHandler<1, spacedim> &dof_handler,
-                    const unsigned int                         obj_level,
-                    const unsigned int                         obj_index,
-                    const unsigned int                         fe_index,
-                    const unsigned int                         local_index,
-                    const std::integral_constant<int, 1> &,
+      set_dof_index(const DoFHandlerType &dof_handler,
```

Comment:
The functions `set_dof_index()`, `get_dof_index()`, `n_active_fe_indices()`, `nth_active_fe_index()`, `get_active_fe_indices`, and `fe_index_is_active()` are the functions accessing the internal CRS data structures of `DoFHandler`. Their implementation are completely new and so that a thorough review here would be good.

The nice aspect that no specialization for different dim/spacedim/structdim is necessary!